### PR TITLE
[DONE] US276433 : Parity for platform services with GitLab

### PR DESCRIPTION
**Pull request by SiniWilson**

### CODE REVIEW

File: installscripts/jazz-terraform-unix-noinstances/scripts/scmpush.sh [hunk 1]
```diff
@@ -11,7 +11,13 @@ ns_id_slf=$6
 
 # Config variables
 scm=$7
-jazzbuildmodule=$8
+jenkins_elb=$8
+jenkins_user=$9
+jenkins_password=${10}
+jazzbuildmodule=${11}
+job_url="/project/Trigger_Platform_Services_Build"
```
Comment (suryajak) [inline installscripts/jazz-terraform-unix-noinstances/scripts/scmpush.sh]: this variable is very specific to gitlabs, so can we name it appropriately?

File: installscripts/jazz-terraform-unix-noinstances/scripts/modifyCodebase.sh [hunk 2]
```diff
@@ -61,15 +61,15 @@ do
 			  "SERVICE_RUNTIME":{"S":"nodejs"},
 			  "SERVICE_STATUS":{"S":"active"},
 			  "TIMESTAMP":{"S":"'$timestamp'"},
+			  "SERVICE_TYPE":{"S":"lambda"},
```
Comment (suryajak) [inline installscripts/jazz-terraform-unix-noinstances/scripts/modifyCodebase.sh]: Please use separate PR's for respective stories. That way its easier for us to validate, track and manage our stories.

File: installscripts/cookbooks/jenkins/files/jobs/job_trigger_platform_services_build.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
+JENKINS_URL=http://$1/ # localhost or jenkins elb url
+JOB_NAME="Trigger_Platform_Services_Build"
```
Comment (suryajak) [inline installscripts/cookbooks/jenkins/files/jobs/job_trigger_platform_services_build.sh]: This is specific to gitlabs, so name it appropriately. Perhaps we should also try to separate code specific to each SCM.

File: installscripts/jazz-terraform-unix-noinstances/scripts/scmpush.sh [hunk 4]
```diff
@@ -48,7 +47,7 @@ function individual_repopush() {
     # Creating the repo in SLF folder in SCM
     repo_id=$(curl -sL --header "PRIVATE-TOKEN: $token" -X POST "http://$scmelb/api/v4/projects?name=$1&namespace_id=$ns_id_slf" | awk -F',' '{print $1}'| awk -F':' '{print $2}')
     #Adding webhook to the repo
-	curl --header "PRIVATE-TOKEN: $token" -X POST "http://$scmelb/api/v3/projects/$repo_id/hooks?enable_ssl_verification=false&push_events=true&url=$gitlab_webhook_url"
+	curl --header "PRIVATE-TOKEN: $token" -X POST "http://$scmelb/api/v4/projects/$repo_id/hooks?enable_ssl_verification=false&push_events=true&url=$gitlab_webhook_url"
```
[the remark this answers is not in the document]
Comment (Feroz-Shaikh) [reply installscripts/jazz-terraform-unix-noinstances/scripts/scmpush.sh]: @suryajak @SiniWilson 
Gitlab API v3 will soon be unsupported as per the official docs. Please refer to this link for more details on API versions: https://docs.gitlab.com/ce/api/v3_to_v4.html
I have added API v4 in all the installer-side code. I guess Daiemen & Pai started with API v3 in the core repo, which looking at the docs, they shouldn't ideally for future compatibility. I believe we should be moving towards the new API v4, which supports almost all the v3 endpoints.

File: installscripts/jazz-terraform-unix-noinstances/scripts/modifyCodebase.sh (hunk 2 above)
Comment (SiniWilson) [reply installscripts/jazz-terraform-unix-noinstances/scripts/modifyCodebase.sh]: going forward i will take care of this

File: installscripts/cookbooks/jenkins/files/jobs/job_trigger_platform_services_build.sh (hunk 3 above)
Comment (SiniWilson) [reply installscripts/cookbooks/jenkins/files/jobs/job_trigger_platform_services_build.sh]: In cooperated the comment

File: installscripts/jazz-terraform-unix-noinstances/scripts/scmpush.sh (hunk 1 above)
Comment (SiniWilson) [reply installscripts/jazz-terraform-unix-noinstances/scripts/scmpush.sh]: In cooperated the comment

**Pull request merged**